clients/python/src/model_registry/_experiments.py

```diff
@@ -0,0 +1,204 @@
+import time
+from contextlib import AbstractContextManager
+from dataclasses import dataclass
+from typing import Any, Callable, Literal
+
+from model_registry.core import ModelRegistryAPIClient
+from model_registry.exceptions import StoreError
+from model_registry.types.artifacts import (
+    ArtifactState,
+    DataSet,
+    ExperimentRunArtifact,
+    ExperimentRunArtifactTypes,
+    Metric,
+    Parameter,
+    ParameterType,
+)
+from model_registry.types.experiments import ExperimentRun
+
+from .utils import S3Params, ThreadSafeVariable, upload_to_s3
+
+LogType = Literal["params", "metrics", "datasets"]
+
+
+@dataclass
+class RunContext:
+    id: str | None = None
+    name: str | None = None
+    run_id: str | None = None
```

Comment:
I believe this is one example of | syntax which is complaining not passing the ci/GHA